sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpEventHubClient.cs

```diff
@@ -0,0 +1,421 @@
+        /// <param name="connectionScope">The optional scope to use for AMQP connection management.  If <c>null</c>, a new scope will be created.</param>
+        /// <param name="messageConverter">The optional converter to use for transforming AMQP message-related types.  If <c>null</c>, a new converter will be created.</param>
+        ///
+        /// <remarks>
+        ///   As an internal type, this class performs only basic sanity checks against its arguments.  It
+        ///   is assumed that callers are trusted and have performed deep validation.
+        ///
+        ///   Any parameters passed are assumed to be owned by this instance and safe to mutate or dispose;
+        ///   creation of clones or otherwise protecting the parameters is assumed to be the purview of the
+        ///   caller.
+        /// </remarks>
+        ///
+        protected AmqpEventHubClient(string host,
+                                     string eventHubName,
+                                     TokenCredential credential,
+                                     EventHubClientOptions clientOptions,
+                                     EventHubRetryPolicy defaultRetryPolicy,
+                                     AmqpConnectionScope connectionScope,
+                                     AmqpMessageConverter messageConverter)
+        {
+            Guard.ArgumentNotNullOrEmpty(nameof(host), host);
+            Guard.ArgumentNotNullOrEmpty(nameof(eventHubName), eventHubName);
+            Guard.ArgumentNotNull(nameof(credential), credential);
+            Guard.ArgumentNotNull(nameof(clientOptions), clientOptions);
+            Guard.ArgumentNotNull(nameof(defaultRetryPolicy), defaultRetryPolicy);
+
+            try
+            {
+                EventHubsEventSource.Log.EventHubClientCreateStart(host, eventHubName);
+
+                _retryPolicy = defaultRetryPolicy;
+                _tryTimeout = _retryPolicy.CalculateTryTimeout(0);
+
+                EventHubName = eventHubName;
+                Credential = credential;
+                MessageConverter = messageConverter ?? new AmqpMessageConverter();
+
+                if (connectionScope == null)
+                {
+                    var endpointBuilder = new UriBuilder
+                    {
+                        Scheme = clientOptions.TransportType.GetUriScheme(),
+                        Host = host
+                    };
+
+                    connectionScope = new AmqpConnectionScope(endpointBuilder.Uri, eventHubName, credential, clientOptions.TransportType, clientOptions.Proxy);
+                }
+
+                ConnectionScope = connectionScope;
+                ManagementLink = new FaultTolerantAmqpObject<RequestResponseAmqpLink>(timeout => ConnectionScope.OpenManagementLinkAsync(timeout, CancellationToken.None), link => link.SafeClose());
```

Comment:
I regret the lack of cancellation token here, but could not devise a means to safely and deterministically capture the token that may be in scope when this is called at arbitrary points.  Given the lack of cancellation token support in the AMQP library overall, it doesn't cut against the pattern of needing to manually check too badly.